MLH-Fellowship/Social-BERTerfly
# Add Twitter predictions

Should solve #10 .

### Changes made:

- Made necessary modifications to `twitterscraper.py`.
- Changed cleaning functions
- Ran batch prediction on 50 latest tweets.
- Created a new form field in `index.html` for entering twitter handle.
- Created a new function `predict_tweet` for batch prediction on 50 tweets.
- Return the mode of the index arrays predicted on 50 tweets.
- Print result to terminal.

### How to test:

Follow a similar approach to when you were entering normal text, instead just enter the twitter handle. Make sure your `.env` file is set up with the correct credentials. Give the server some time at the start to boot up and load the model. The response json should be printing in the following form:
`{"type":<personality_type>}`.

### Challenges:

I believe the kaggle dataset has a large bias towards Introvert personality types. Hence a lot of my tries gave me introvert personality types. We can try playing around with the no. of tweets possibly, but I need to know the API request limit for the Twitter API we're using.
